extensions/vertx-http/deployment/src/main/java/io/quarkus/vertx/http/deployment/HttpRootPathBuildItem.java

```diff
@@ -1,26 +1,58 @@
 package io.quarkus.vertx.http.deployment;
 
+import java.net.URI;
+
 import io.quarkus.builder.item.SimpleBuildItem;
+import io.quarkus.deployment.util.UriNormalizationUtil;
 
 public final class HttpRootPathBuildItem extends SimpleBuildItem {
 
-    private final String rootPath;
+    /**
+     * Normalized from quarkus.http.root-path.
+     * This path will always end in a slash
+     */
+    private final URI rootPath;
 
     public HttpRootPathBuildItem(String rootPath) {
-        this.rootPath = rootPath;
+        this.rootPath = UriNormalizationUtil.toURI(rootPath, true);
     }
 
+    /**
+     * Return normalized Http root path configured from {@literal quarkus.http.root-path}.
+     * This path will always end in a slash.
+     * <p>
+     * Use {@link #resolvePath(String)} if you need to construct a Uri from the Http root path.
+     *
+     * @return Normalized Http root path ending with a slash
+     * @see #resolvePath(String)
+     */
     public String getRootPath() {
-        return rootPath;
+        return rootPath.getPath();
     }
 
-    public String adjustPath(String path) {
```

Comment:
this removal is breaking quarkus-qe/quarkus-openshift-test-suite

It can break other users who build their own extensions and use HttpRootPathBuildItem.

https://github.com/quarkus-qe/quarkus-openshift-test-suite/blob/master/app-metadata/deployment/src/main/java/io/quarkus/ts/openshift/app/metadata/AppMetadataCollector.java#L35

It should be at least documented what's the replacement using `resolvePath` method.
Better option would be to deprecate the method and keep it there for some time, `adjustPath` could call `resolvePath` if possible.

Reply:
We put it back, with deprecation markers. 

In general, you just shouldn't munge the path yourself.. there are too many configurations to get right.